mount_test.go

```diff
@@ -41,3 +49,132 @@ func TestIsSystemMount(t *testing.T) {
 		}
 	}
 }
+
+func TestMajorMinorNumber(t *testing.T) {
+	devices := []string{"/dev/zero", "/dev/net/tun"}
+
+	for _, device := range devices {
+		cmdStr := fmt.Sprintf("ls -l %s | awk '{print $5$6}'", device)
+		cmd := exec.Command("sh", "-c", cmdStr)
+		output, err := cmd.Output()
+
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		data := bytes.Split(output, []byte(","))
+		if len(data) < 2 {
+			t.Fatal()
+		}
+
+		majorStr := strings.TrimSpace(string(data[0]))
+		minorStr := strings.TrimSpace(string(data[1]))
+
+		majorNo, err := strconv.Atoi(majorStr)
+		minorNo, err := strconv.Atoi(minorStr)
+
+		stat := syscall.Stat_t{}
+		err = syscall.Stat(device, &stat)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Get major and minor numbers for the device itself. Note the use of stat.Rdev instead of Dev.
+		major := major(stat.Rdev)
+		minor := minor(stat.Rdev)
+
+		if minor != minorNo {
+			t.Fatalf("Expected minor number for device %s: %d, Got :%d", device, minorNo, minor)
+		}
+
+		if major != majorNo {
+			t.Fatalf("Expected major number for device %s : %d, Got :%d", device, majorNo, major)
+		}
+	}
+}
+
+func TestGetDeviceForPathRoot(t *testing.T) {
+	dev, err := getDeviceForPath("/")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := "/"
+
+	if dev.mountPoint != expected {
+		t.Fatalf("Expected %s mountpoint, got %s", expected, dev.mountPoint)
+	}
+}
+
+func TestGetDeviceForPathSuccess(t *testing.T) {
+	dev, err := getDeviceForPath("/proc")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := "/proc"
+
+	if dev.mountPoint != expected {
+		t.Fatalf("Expected %s mountpoint, got %s", expected, dev.mountPoint)
+	}
+}
+
+func TestGetDeviceForPathEmptyPath(t *testing.T) {
+	_, err := getDeviceForPath("")
```

Comment:
I think there is scope to add some more tests for this function:

- `getDeviceForPath("///")`
- `getDeviceForPath("/../../.././././../.")`
- what if the specified path has spaces / special characters / unicode?
- create a test that creates a bind mount manually and see if `getDevicePath()` returns the expected value.